### PR TITLE
Update HttpClient collector to the latest specification

### PR DESCRIPTION
**Pull request by alexvaluyskiy**
Refactored attributes based on the latest spec: https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/data-http.md

Added attributes
- `component`
- `http.flavor`
- ~~`http.scheme`~~
- ~~`http.status_text`~~

Removed attributes
- `http.request.size` - never used
- `http.response.size` - never used
- `http.user_agent`
- `http.path`

### CODE REVIEW

**Comment by alexvaluyskiy** [comment]
I signed it

**Comment by alexvaluyskiy** [comment]
I signed it

**Comment by alexvaluyskiy** [comment]
I signed it

File: src/OpenTelemetry.Collector.Dependencies/Implementation/HttpHandlerDiagnosticListener.cs [hunk 1]
```diff
@@ -69,12 +69,13 @@ public override void OnStartActivity(Activity activity, object payload)
 
             if (span.IsRecording)
             {
+                span.PutComponentAttribute("http");
+                span.PutHttpTargetAttribute(request.RequestUri.PathAndQuery);
```
Comment (lmolkova) [inline src/OpenTelemetry.Collector.Dependencies/Implementation/HttpHandlerDiagnosticListener.cs]: spec requires only one set

```
One of the following sets of attributes is required (in order of usual preference unless for a particular web client/framework it is known that some other set is preferable for some reason; all strings must be non-empty):

http.url
http.scheme, http.host, http.target
http.scheme, peer.hostname, peer.port, http.target
http.scheme, peer.ip, peer.port, http.target
```

why should we populate target, flavor and scheme? They all could be inferred from url and introduce redunduncy.

File: src/OpenTelemetry.Collector.Dependencies/Implementation/HttpHandlerDiagnosticListener.cs [hunk 2]
```diff
@@ -69,12 +69,13 @@ public override void OnStartActivity(Activity activity, object payload)
...
                 span.PutHttpMethodAttribute(request.Method.ToString());
                 span.PutHttpHostAttribute(request.RequestUri.Host, request.RequestUri.Port);
-                span.PutHttpPathAttribute(request.RequestUri.AbsolutePath);
-                request.Headers.TryGetValues("User-Agent", out var userAgents);
-                span.PutHttpUserAgentAttribute(userAgents?.FirstOrDefault());
                 span.PutHttpRawUrlAttribute(request.RequestUri.OriginalString);
+                span.PutHttpSchemeAttribute(request.RequestUri.Scheme);
+                span.PutHttpFlavorAttribute(request.Version.ToString());
```
Comment (lmolkova) [inline src/OpenTelemetry.Collector.Dependencies/Implementation/HttpHandlerDiagnosticListener.cs]: version is not required, why are we populating it?

File: src/OpenTelemetry.Api/Trace/SpanExtensions.cs [hunk 3]
```diff
@@ -170,6 +157,10 @@ public static ISpan PutHttpRequestSizeAttribute(this ISpan span, long size)
         public static ISpan PutHttpStatusCode(this ISpan span, int statusCode, string reasonPhrase)
         {
             span.PutHttpStatusCodeAttribute(statusCode);
+            if (!string.IsNullOrEmpty(reasonPhrase))
+            {
+                span.SetAttribute(SpanAttributeConstants.HttpStatusTextKey, reasonPhrase);
```
Comment (lmolkova) [inline src/OpenTelemetry.Api/Trace/SpanExtensions.cs]: it is not required. why do we need to set it?

File: test/OpenTelemetry.Collector.Dependencies.Tests/HttpClientTests.cs [hunk 4]
```diff
@@ -170,18 +170,22 @@ public async Task DebugIndividualTestAsync()
...
     ""spanKind"": ""Client"",
     ""spanAttributes"": {
-      ""http.path"": ""/"",
+      ""component"": ""http"",
       ""http.method"": ""GET"",
+      ""http.url"": ""http://{host}:{port}/path/12314/?q=ddds#123"",
+      ""http.target"": ""/path/12314/?q=ddds"",
       ""http.host"": ""{host}:{port}"",
+      ""http.scheme"": ""http"",
+      ""http.flavor"": ""2.0"",
```
Comment (lmolkova) [inline test/OpenTelemetry.Collector.Dependencies.Tests/HttpClientTests.cs]: is there a way to specify that this and few others are optional?

File: src/OpenTelemetry.Collector.Dependencies/Implementation/HttpHandlerDiagnosticListener.cs (hunk 1 above)
Comment (alexvaluyskiy) [reply src/OpenTelemetry.Collector.Dependencies/Implementation/HttpHandlerDiagnosticListener.cs]: Ok, I will remove `http.target` and `http.scheme`

**Comment by alexvaluyskiy** [comment]
We could enable all optional attributes via configuration. For example, to add two properties to `HttpClientCollectorOptions`
- `LogHttpVersion`
- `LogStatusText`

What do you think?

**Comment by lmolkova** [comment]
> We could enable all optional attributes via configuration. 

good point! Can we name them 'SetHttpFlavor' and `SetStatusText`? And keep them false by default.

**Comment by alexvaluyskiy** [comment]
Sure

**Comment by alexvaluyskiy** [comment]
@lmolkova I’ve removed redundant attributes and have added `http.flavor` into the options

**Pull request merged**